pkg/nmi/server/endpointpolicy.go

```diff
@@ -0,0 +1,175 @@
+// +build windows
+
+package server
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	v1 "github.com/Microsoft/hcsshim"
+	msg "github.com/Microsoft/hcnproxy/pkg/types"
+	client "github.com/Microsoft/hcnproxy/pkg/client"
+	"k8s.io/klog"
+)
+
+// ApplyEndpointRoutePolicy applies the route policy against the pod ip endpoint
+func ApplyEndpointRoutePolicy(podIP string, metadataIP string, metadataPort string, nmiIP string, nmiPort string) error {
+	if len(podIP) <= 0 {
+		return errors.New("Missing IP Address")
+	}
+
+	endpoint, err := getEndpointByIP(podIP)
+	if err != nil {
+		return err
+	}
+
+	er := addEndpointPolicy(endpoint, metadataIP, metadataPort, nmiIP, nmiPort)
+	if er != nil {
+		return err
+	}
+
+	return nil
+}
+
+// DeleteEndpointRoutePolicy applies the route policy against the pod ip endpoint
+func DeleteEndpointRoutePolicy(podIP string, metadataIP string) error {
+	if len(podIP) <= 0 {
+		return errors.New("Missing IP Address")
+	}
+
+	endpoint, err := getEndpointByIP(podIP)
+	if err != nil {
+		return err
+	}
+
+	er := deleteEndpointPolicy(endpoint, metadataIP)
+	if er != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getEndpointByIP(ip string) (*v1.HNSEndpoint, error) {
+	klog.Infof("Getting endpoint for IP %s\n", ip)
+
+	request := msg.HNSRequest{
+		Entity:    msg.EndpointV1,
+		Operation: msg.Enumerate,
+		Request:   nil,
+	}
+
+	klog.Info("Enumerating all endpoints\n")
+	response, e := callHcnProxyAgent(request)
```

Comment:
>e [](start = 11, length = 1)

Why do you use different error variable names throughout the code? `e, er, err` 
What each name means and why there are many?